File: templates/src/pkg/params.ts
/** Parameter composition: apply BC_PAR_* env overrides to the threaded opts. */
import type { Opts } from "big-config";
import * as bcWorkflow from "big-config/workflow";
import { syncAliases, toBcOpts } from "./interop.js";

/** Per-stage opts function: fold BC_PAR_* overrides into params.
 *
 * Grow this to extract params from earlier stages' outputs as you add stages. */
export function optsFn(opts: Opts): Opts {
  return syncAliases(bcWorkflow.readBcPars(toBcOpts(opts)));
}

File: create-py-bigconfig-package/templates/src/pkg/params.py
"""Parameter composition: apply BC_PAR_* env overrides to the threaded opts."""
from __future__ import annotations

from big_config.core import Opts

from .interop import read_bc_pars, sync_aliases


def opts_fn(opts: Opts) -> Opts:
    """Per-stage opts function: fold BC_PAR_* overrides into params.

    Grow this to extract params from earlier stages' outputs as you add stages.
    """
    return sync_aliases(read_bc_pars(opts))


